src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ChainPal.cs

```diff
@@ -35,10 +35,10 @@ public static bool ReleaseSafeX509ChainHandle(IntPtr handle)
             TimeSpan timeout,
             bool disableAia)
         {
-            // An input value of 0 on the timeout is "take all the time you need".
+            // An input value of 0 on the timeout is treated as 15 seconds, to match Windows.
             if (timeout == TimeSpan.Zero)
             {
-                timeout = TimeSpan.MaxValue;
+                timeout = TimeSpan.FromSeconds(15);
             }
 
```

Comment:
Shouldn't there also be an "if it's bigger than 15 seconds, clamp it"?

Reply:
Eh I was waiting on "go ahead" from #40578, but I guess this is breaking enough as it is.

Reply:
I don't see the upper bound clamping behavior as having been applied.

I think I'm OK with making the policy property setter throw for negative inputs (but have pinged @terrajobst offline for a conscience/reality check)... so maybe we don't need to deal with negative numbers here.

Reply:
Just as I said what I said about the negative numbers he convinced me that any mistakes in the wild have been accruing for 15 years (since .NET Framework 2.0); so we should just stick with "all negative numbers mean one minute".

Reply:
Okay, I am not convinced that Windows' max is 1 minute, that is the behavior I have observed so far. Regardless, negative values "work" on Windows, so we can at least make it work on Linux with a 1 minute timeout.